base/range.jl

```diff
@@ -244,6 +244,52 @@ function show(io::IO, r::LinSpace)
     print(io, ')')
 end
 
+# print_range() prints out a nice looking range in terms of its elements
+# as if it were collect(range), dependent on the size of the
+# terminal, and taking into account whether compact numbers should be shown.
+# It figures out the width in characters of each element, and if they
+# end up too wide, it shows the first and last elements separated by a
+# horizontal elipsis.
+# Optional parameters include sz for the (rows,cols) of the screen,
+# pre and post characters for each printed row, a separator string sep between
+# printed elements, string for the horizontal ellipsis.
+# typical output will look like this (for a narrow screen):
+#  1.0,2.0,3.0,…,4.0,5.0,6.0
+# This function is borrowed from print_matrix in show.jl
+# It is usually called by writemime (from replutil.jl)
+function print_range(io::IO, r::Union{Range,LinSpace},
```

Comment:
This `Union` is unnecessary (similarly in `writemime`, below), because `LinSpace <: Range`.